superdesk/publish/formatters/__init__.py

```diff
@@ -135,6 +136,38 @@ def set_destination(self, destination=None, subscriber=None):
         self.destination = destination
         self.subscriber = subscriber
 
+    def remove_all_embeds(self, article):
+        """
+        Removes any embeds from the draftjs state and regenerates the html, can be used by text only
+        formatters to remove embeds from the article
+        :param article:
+        :return:
+        """
+
+        # List of keys of the removed entities
+        keys = []
+
+        def not_embed(block):
+            if block.type.lower() == "atomic":
+                keys.extend([e.key for e in block.entities])
+                for idx, kk in enumerate(block.entities):
+                    del block.entities[idx]
+                del block.entities
```

Comment:
When a `Block` instance is constructed, the `entites` attribute is populated (https://github.com/superdesk/superdesk-core/blob/1050960444d26b9d845f0ecde78b910782c6e01e/superdesk/editor_utils.py#L186C25-L186C39).

As to ensure the `entities` attribute is always defined, would it be good to add a `clear` function in the `EntitySequence` class and use it here instead?

Reply:
Done